modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +108,71 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm.resize(A.n);
+    iperm.resize(A.n);
+
+    if(computePermutation)
```

Comment:
Are there any cases where the permutation must be updated? For example, when the matrix size changes, when the matrix structure changes. If yes, it needs to be checked.